### PR TITLE
code review

### PR DESCRIPTION
**Pull request by ccoVeille**
- **chore: remove unused parameters**
- **chore: fix Go comments**
- **chore: use simpler syntax**
- **chore: remove useless method**
- **chore: avoid type juggling**

- **chore: reduce complexity**

I used golangci with a lot of settings to detect many of the issues I report here

### CODE REVIEW

File: processor/processor.go [hunk 1]
```diff
@@ -332,7 +332,7 @@ func setupCountAs() {
 			// See if we can identify based on language name which is the most
 			// reliable as the name should be unique
 			for name := range languageDatabase {
-				if strings.ToLower(name) == strings.ToLower(t[1]) {
+				if strings.EqualFold(name, t[1]) {
```
Comment (ccoVeille) [inline processor/processor.go:335]: Note for myself, remove this

See https://github.com/boyter/scc/pull/479#discussion_r1632882275

cc @apocelipes

Comment (ccoVeille) [reply processor/processor.go]: I removed in favor of #479

**Comment by boyter** [comment]
Annoyingly there is a conflict. If you want to update ill be happy to merge once its resolved.

**Comment by ccoVeille** [comment]
Sure

**Pull request merged**